icaruscode/CRT/CRTTPCTruthEffMCKalman_module.cc

```diff
@@ -0,0 +1,1721 @@
+
+			crt_tpc_plane_dist = CRT_plane_dist(trkpos,trkdir,crtpos,thinrange);
+
+		}//end if(best_dca_pos==0||best_dca_pos==2)
+		else if(best_dca_pos==1){
+
+			TVector3 trkpos(closest.tpc_track_end.X(),closest.tpc_track_end.Y(),closest.tpc_track_end.Z());
+			TVector3 trkdir(closest.endDir.X(),closest.endDir.Y(),closest.endDir.Z());
+			TVector3 crtpos(closest.thishit.x_pos,closest.thishit.y_pos,closest.thishit.z_pos);
+
+			crt_tpc_plane_dist = CRT_plane_dist(trkpos,trkdir,crtpos,thinrange);
+
+		}//end else if(best_dca_pos==1)
+
+		crttime = closest.t0;
+		crt_pes= closest.thishit.peshit;
+		crt_extraplen = closest.extrapLen;
+
+		tpc_trk_start_x = closest.tpc_track_start.X();
+		tpc_trk_start_y = closest.tpc_track_start.Y();
+		tpc_trk_start_z = closest.tpc_track_start.Z();
+
+		tpc_trk_end_x = closest.tpc_track_end.X();
+		tpc_trk_end_y = closest.tpc_track_end.Y();
+		tpc_trk_end_z = closest.tpc_track_end.Z();
+	
+		if((tpc_trk_start_x+tpc_trk_end_x)/2 >0) {
+
+			tpcWest=true;
+
+		}//end if((tpc_trk_start_x+tpc_trk_end_x)/2 >0) 
+		else if((tpc_trk_start_x+tpc_trk_end_x)/2 <0){
+
+			tpcEast=true;
+
+		}//end else if((tpc_trk_start_x+tpc_trk_end_x)/2 <0)
+
+		trk_startdir_x = closest.startDir.X();
+		trk_startdir_y = closest.startDir.Y();
+		trk_startdir_z = closest.startDir.Z();
+	
+		trk_enddir_x = closest.endDir.X();
+		trk_enddir_y = closest.endDir.Y();
+		trk_enddir_z = closest.endDir.Z();
+
+		crt_x = closest.thishit.x_pos;
+		crt_y = closest.thishit.y_pos;
+		crt_z = closest.thishit.z_pos;
+
+		t0min = closest.t0min;
+		t0max = closest.t0max;
+
+		PCA_DCA_start = closest.PCA_DCA_start;
+		PCA_DCA_end = closest.PCA_DCA_end;
+
+		crt_hitflag = closest.thishit.ts0_s_corr;
+
+		//extract all info about CRT Hit's mac5+channel+pes into vectors
+	    	std::map< uint8_t, std::vector<std::pair<int,float> > > crthit_pes = closest.thishit.pesmap; ///< Saves signal hit information (FEB, local-channel and PE) .
+		for(uint8_t i=1; i<232; i++){
+			if(crthit_pes[i].size()>0){
+				for(int j=0; j<(int)crthit_pes[i].size(); j++){
+					matched_hit_mac5.push_back(i);
+					matched_hit_chan.push_back(crthit_pes[i][j].first);
+					matched_hit_pes.push_back(crthit_pes[i][j].second);
+				}//end for(int j=0; j<crt_pes[i].size(); j++)
+			}//end if(crthit_pes[i].size()>0)
+		}//end loop searching for FEBs connected to the hit
+
+ 		std::vector<icarus::match_geometry> all_crt_candidates = t0Alg.GetClosestCRTHit_geo(detProp, *trackList[track_i], hits, crtHits, m_trigger_timestamp, fIsData);
+
+			auto checkhit = closest.thishit;
+			crt_trueID =  bt.TrueIdFromTotalEnergy(event, checkhit);
+			GetAncestorID(crt_trueID, crt_mother, crt_ancestor, crt_motherlayers, particles);
+			crt_pdg = particles[crt_trueID].PdgCode();
+			crt_ancestor_pdg = particles[crt_ancestor].PdgCode();
+
+		has_crtmatch=false;
+		num_crt_candidates = (int)all_crt_candidates.size();
+		if(num_crt_candidates>0){
```

Comment:
```suggestion
		if (num_crt_candidates == 0) continue;
```
This will help "denest" the following block since there are a lot of nested indentations below.